sql/src/test/java/io/crate/metadata/doc/DocIndexMetaDataTest.java

```diff
@@ -679,6 +680,39 @@ public void testExtractMultiplePrimaryKeys() throws Exception {
         assertThat(md.primaryKey(), hasItems(ColumnIdent.fromPath("id"), ColumnIdent.fromPath("title")));
     }
 
+    @Test
+    public void testExtractCheckConstraints() throws Exception {
+        XContentBuilder builder = XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject(Constants.DEFAULT_MAPPING_TYPE)
+            .startObject("_meta")
+            .startObject("check_constraints")
+            .field("test3_check_1", "id >= 0")
+            .field("test3_check_2", "title != 'Pink Power Ranger'")
+            .endObject()
+            .endObject()
+            .startObject("properties")
+            .startObject("id").field("type", "integer").endObject()
+            .startObject("title").field("type", "string").endObject()
+            .endObject()
+            .endObject()
+            .endObject();
+        IndexMetaData metaData = getIndexMetaData("test3", builder);
+        DocIndexMetaData md = newMeta(metaData, "test3");
+        assertThat(md.checkConstraints().size(), is(2));
+        for (AnalyzedCheckConstraint chk : md.checkConstraints()) {
+            switch (chk.name()) {
+                case "test3_check_1":
+                    assertThat(chk.expression(), is(SqlParser.createExpression("id >= 0")));
+                    break;
+
+                case "test3_check_2":
+                    assertThat(chk.expression(), is(SqlParser.createExpression("title <> 'Pink Power Ranger'")));
+                    break;
+            }
+        }
+    }
```

Comment:
Maybe work with `assertThat(md.checkConstraints(), Matchers.contains(...))` gives better error messages.

Reply:
I did it this way to avoid having to know the order of the checks, but I will play with this.

Reply:
There is also a `Matchers.containsInAnyOrder`